### PR TITLE
Disable code coverage redesign in CI

### PR DESCRIPTION
**Pull request by jhendrixMSFT**
Go 1.22 changed some code coverage internals which is causing some cases to under-report coverage. Disabling for now until this is resolved.

See https://github.com/golang/go/issues/65570

### CODE REVIEW

File: eng/pipelines/templates/steps/build-test.yml [hunk 1]
```diff
@@ -80,6 +80,7 @@ steps:
       ${{ insert }}: ${{ parameters.EnvVars }}
       GOTRACEBACK: all
       AZURE_SDK_GO_LOGGING: all
+      GOEXPERIMENT: nocoverageredesign
```
Comment (jhendrixMSFT) [inline eng/pipelines/templates/steps/build-test.yml:83]: Apparently we must set this only for Go 1.22.

**Comment by jhendrixMSFT** [comment]
It also appears to affect the Go version string used in the `User-Agent` header which is causing some tests to fail.

`azsdk-go-azservicebus/v1.0.0 (go1.22.0 X:nocoverageredesign; Windows_NT)`

**Review by weshaggard — APPROVED** [review]

**Pull request merged**